src/Uno.UI/Microsoft/UI/Xaml/Controls/TeachingTip/TeachingTip.cs

```diff
@@ -0,0 +1,2565 @@
+				availability[TeachingTipPlacementMode.BottomLeft] = false;
+			}
+			// If the tip is too wide to fit in the window when the tail is centered horizontally on the target and the tip.
+			if (contentWidth / 2.0f > availableBoundsAroundTarget.Left + (m_currentTargetBoundsInCoreWindowSpace.Width / 2.0f) ||
+				contentWidth / 2.0f > availableBoundsAroundTarget.Right + (m_currentTargetBoundsInCoreWindowSpace.Width / 2.0f))
+			{
+				availability[TeachingTipPlacementMode.Top] = false;
+				availability[TeachingTipPlacementMode.Bottom] = false;
+				availability[TeachingTipPlacementMode.Center] = false;
+			}
+			// If the tip is too wide to fit between the center of the target and the right edge of the window.
+			if (contentWidth - MinimumTipEdgeToTailCenter() > availableBoundsAroundTarget.Right + (m_currentTargetBoundsInCoreWindowSpace.Width / 2.0f))
+			{
+				availability[TeachingTipPlacementMode.TopRight] = false;
+				availability[TeachingTipPlacementMode.BottomRight] = false;
+			}
+			// If the tip is too wide to fit between the right edge of the target and the right edge of the window.
+			if (tipWidth > availableBoundsAroundTarget.Right)
+			{
+				availability[TeachingTipPlacementMode.Right] = false;
+				availability[TeachingTipPlacementMode.RightTop] = false;
+				availability[TeachingTipPlacementMode.RightBottom] = false;
+			}
+
+			var wantedDirection = GetFlowDirectionAdjustedPlacement(PreferredPlacement);
+			var priorities = GetPlacementFallbackOrder(wantedDirection);
+
+			foreach (var mode in priorities)
+			{
+				if (availability[mode])
+				{
+					return Tuple.Create(mode, false);
+				}
+			}
+			// The teaching tip wont fit anywhere, set tipDoesNotFit to indicate that we should not open.
+			return Tuple.Create(TeachingTipPlacementMode.Top, true);
+		}
+
+		Tuple<TeachingTipPlacementMode, bool> DetermineEffectivePlacementUntargeted(double contentHeight, double contentWidth)
+		{
+			var windowBounds = GetWindowBounds();
+			if (!ShouldConstrainToRootBounds)
+			{
+				var screenBoundsInCoreWindowSpace = GetEffectiveScreenBoundsInCoreWindowSpace(windowBounds);
+				if (screenBoundsInCoreWindowSpace.Height > contentHeight && screenBoundsInCoreWindowSpace.Width > contentWidth)
+				{
+					return Tuple.Create(TeachingTipPlacementMode.Bottom, false);
+				}
+			}
+			else
+			{
+				var windowBoundsInCoreWindowSpace = GetEffectiveWindowBoundsInCoreWindowSpace(windowBounds);
+				if (windowBoundsInCoreWindowSpace.Height > contentHeight && windowBoundsInCoreWindowSpace.Width > contentWidth)
+				{
+					return Tuple.Create(TeachingTipPlacementMode.Bottom, false);
+				}
+			}
+
+			// The teaching tip doesn't fit in the window/screen set tipDoesNotFit to indicate that we should not open.
+			return Tuple.Create(TeachingTipPlacementMode.Top, true);
+		}
+
+		private Tuple<Thickness, Thickness> DetermineSpaceAroundTarget()
+		{
+			var shouldConstrainToRootBounds = ShouldConstrainToRootBounds;
```

Comment:
Codacy found an issue: [Remove the unused local variable 'shouldConstrainToRootBounds'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6388179)